boards/PDM/Src/App/App_InRangeCheck.c

```diff
@@ -0,0 +1,58 @@
+#include <stdlib.h>
+#include <assert.h>
+
+#include "App_InRangeCheck.h"
+
+struct InRangeCheck
+{
+    float (*get_value)(void);
+    float min_value;
+    float max_value;
+};
+
+struct InRangeCheck *App_InRangeCheck_Create(
+    float (*const get_value)(void),
+    float min_value,
+    float max_value)
+{
+    assert(get_value != NULL);
+
+    struct InRangeCheck *in_range_check = malloc(sizeof(struct InRangeCheck));
+    assert(in_range_check != NULL);
+
+    in_range_check->get_value = get_value;
+    in_range_check->min_value = min_value;
+    in_range_check->max_value = max_value;
+
+    return in_range_check;
+}
+
+void App_InRangeCheck_Destroy(struct InRangeCheck *const in_range_check)
+{
+    free(in_range_check);
+}
+
+enum InRangeCheck_Status App_InRangeCheck_GetValue(
```

Comment:
I ended up removing the callback functions for `InRangeCheck`  because it was quite hard to test (maybe that's a sign it wasn't a good design?). The problem was that the callback functions weren't part of the state machine logic and thus couldn't be tested.